packages/console/src/pages/Profile/index.tsx

```diff
@@ -1,78 +1,49 @@
-import type { IdTokenClaims } from '@logto/react';
-import { useLogto } from '@logto/react';
-import type { Nullable } from '@silverhand/essentials';
-import { useState, useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
 
 import Button from '@/components/Button';
 import CardTitle from '@/components/CardTitle';
-import UserAvatar from '@/components/UserAvatar';
 import { isCloud } from '@/consts/cloud';
+import useLogtoUserInfo from '@/hooks/use-logto-userinfo';
 import * as resourcesStyles from '@/scss/resources.module.scss';
 
-import type { Row } from './components/CardContent';
+import BasicUserInfoSection from './components/BasicUserInfoSection';
 import CardContent from './components/CardContent';
+import PasswordSection from './components/PasswordSection';
 import Section from './components/Section';
 import * as styles from './index.module.scss';
 
 const Profile = () => {
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
-  const { getIdTokenClaims } = useLogto();
-  const [user, setUser] = useState<IdTokenClaims>();
-
-  useEffect(() => {
-    (async () => {
-      const claims = await getIdTokenClaims();
-
-      if (claims) {
-        setUser(claims);
-      }
-    })();
-  }, [getIdTokenClaims]);
+  const [user, fetchUser] = useLogtoUserInfo();
 
   if (!user) {
     return null;
   }
 
-  const { name, username, picture, email } = user;
-
-  const conditionalUsername: Array<Row<Nullable<string> | undefined>> = isCloud
-    ? [{ label: 'profile.settings.username', value: username }]
-    : [];
-
   return (
     <div className={resourcesStyles.container}>
       <div className={resourcesStyles.headline}>
         <CardTitle title="profile.title" subtitle="profile.description" />
       </div>
-      <Section title="profile.settings.title">
-        <CardContent
-          title="profile.settings.profile_information"
-          data={[
-            {
-              label: 'profile.settings.avatar',
-              value: picture,
-              renderer: (value) => <UserAvatar className={styles.avatar} url={value} />,
-            },
-            { label: 'profile.settings.name', value: name },
-            ...conditionalUsername,
-          ]}
-        />
-      </Section>
+      <BasicUserInfoSection user={user} onUpdate={fetchUser} />
       {isCloud && (
         <Section title="profile.link_account.title">
           <CardContent
             title="profile.link_account.email_sign_in"
-            data={[{ label: 'profile.link_account.email', value: email }]}
+            data={[
+              {
+                label: 'profile.link_account.email',
+                value: user.email,
+                actionName: 'profile.link',
+                action: () => {
+                  console.log('link email');
```

Comment:
is this console.log expected?

Reply:
If the content "link email" is expected, should this be an i18n key, and we may refactor the `action` method?

Reply:
This is a placeholder and will be updated in the follow-up PRs.

Reply:
Cool, maybe can use `//TODO: @name LOG-xxxx [description]` formatted comments so that reviewers can get the idea in no time.

Reply:
OK, will do that next time. Thanks for advice